featuretools/tests/primitive_tests/natural_language_primitives_tests/test_natural_language_primitives_terminate.py

```diff
@@ -0,0 +1,39 @@
+import signal
+from platform import system
+
+import pandas as pd
+import pytest
+
+from featuretools.primitives.utils import get_natural_language_primitives
+
+TIMEOUT_THRESHOLD = 20
+
+
+@pytest.mark.skipif(
+    system() == "Linux" or system() == "Darwin",
+    reason="timeout test only supported on UNIX systems",
+)
+@pytest.mark.parametrize(
+    "primitive",
+    get_natural_language_primitives(),
+)
+def test_natlang_primitive_does_not_timeout(
+    strings_that_have_triggered_errors_before, primitive
+):
+    def handle_SIGALRM(signum, frame):
+        raise TimeoutError(
+            f"NaturalLanguage primitive took longer than {TIMEOUT_THRESHOLD}"
+        )
+
+    signal.signal(signal.SIGALRM, handle_SIGALRM)
+    for text in strings_that_have_triggered_errors_before:
+        signal.alarm(TIMEOUT_THRESHOLD)
+        try:
+            primitive().get_function()(pd.Series(text))
+        except TimeoutError:
+            raise TimeoutError(
```

Comment:
if we're going to raise the error anyway, does this need to be a try/except block?